src/framework/xr/xr-anchor.js

```diff
@@ -61,13 +91,37 @@ class XrAnchor extends EventHandler {
      * });
      */
 
+    /**
+     * Fired when an {@link XrAnchor}'s has been persisted.
+     *
+     * @event XrAnchor#persist
+     * @param {string} uuid - Unique string that can be used to restore this anchor.
+     * @example
+     * anchor.on('persist', function (uuid) {
+     *     // anchor has been persisted
+     * });
+     */
+
+    /**
+     * Fired when an {@link XrAnchor}'s has been forgotten.
+     *
+     * @event XrAnchor#forget
+     * @example
+     * anchor.on('forget', function () {
+     *     // anchor has been forgotten
+     * });
+     */
+
     /**
      * Destroy an anchor.
      */
     destroy() {
         if (!this._xrAnchor) return;
         this._anchors._index.delete(this._xrAnchor);
 
+        if (this._uuid)
+            this._anchors._indexByUuid.delete(this._uuid);
```

Comment:
Perhaps this whole (or at least majority of this) function should be moved to the Anchors class, and here we would just call it, to avoid using private properties.